astropy/cosmology/_src/core.py

```diff
@@ -91,7 +93,9 @@ class CosmologyError(Exception):
 class _NameField:
     default: str | None = None
 
-    def __get__(self, instance: Union["Cosmology", None], owner: type) -> str:
+    def __get__(
+        self, instance: Union["Cosmology", None], owner: type["Cosmology"] | None
```

Comment:
We can avoid the `Union` import with
```suggestion
        self, instance: "Cosmology | None", owner: type["Cosmology"] | None
```

Reply:
I was looking up which preferable for partially stringified annotations and chose this. This way `"Cosmology"` is the only string forward reference that needs parsing.
The forward-ref PEP will fix much of this when it lands.

Reply:
that's fine, I don't mind at all.

Reply:
by the way, in case you're not caught up yet: forward ref isn't going to land, at least not as initially planned
see https://docs.python.org/3.14/whatsnew/3.14.html#whatsnew314-pep649
specifically, PEP 563 is now replaced with PEP 649

Reply:
Thanks!